File: back/src/models/PhilopateerRule.js
import mongoose from 'mongoose';

const philopateerRuleSchema = new mongoose.Schema(
  {
    serviceType: { type: String, required: true, unique: true, index: true }, // poster, video, montage, office, photography
    minDaysRequired: { type: Number, default: 3 },
    description: { type: String, default: '' }
  },
  {
    timestamps: true,
    collection: 'philopateer_rules'
  }
);

const PhilopateerRule = mongoose.models.PhilopateerRule || mongoose.model('PhilopateerRule', philopateerRuleSchema);
export default PhilopateerRule;

File: back/Models/PhilopateerRule.cs
using System;
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;

namespace Eftekad.Backend.Models
{
    [Table("philopateer_rules")]
    public class PhilopateerRule
    {
        [Key]
        [Column("id")]
        public string Id { get; set; } = string.Empty;

        [Column("serviceType")]
        public string ServiceType { get; set; } = string.Empty;

        [Column("minDaysRequired")]
        public int MinDaysRequired { get; set; } = 3;

        [Column("description")]
        public string Description { get; set; } = string.Empty;

        [Column("createdAt")]
        public DateTime CreatedAt { get; set; } = DateTime.UtcNow;

        [Column("updatedAt")]
        public DateTime UpdatedAt { get; set; } = DateTime.UtcNow;
    }
}
